README.md

```diff
@@ -8,36 +8,133 @@ Please fork this repository and answer any questions on this markdown document.
+done
+
+
+
 # Docker
 * In the root of this repository, create a Dockerfile that is based on the latest mariadb image.
     * Expose port 3307.
-    * Define an evironment variable called BRUCE with a value of WAYNE.
+    * Define an evironment variable called BRUCE with a value of WAYNE.	
     * Run a command that will output the value from BRUCE into a file called BATCAVE in the root directory of the container. 
+	
+	FROM mariadb:latest
+	EXPOSE 3307
+	ENV BRUCE=WAYNE
+	RUN echo "$(BRUCE) > /root/BATCAVE"
+
+	
+	
 * Create a Bash script in the root of this repository called FLY.sh that will do the following:
     * Install Docker if it is not yet installed.
     * Ensure the installed version of Docker is the latest version available.
+	
+			
+#!/bin/bash
+yum install -y yum-utils device-mapper-persistent-data lvm2
+
+echo "Updating and installing Docker"
+yum-config-manager   --add-repo  https://download.docker.com/linux/centos/docker-ce.repo
+yum install -y docker-ce
+
+echo "Starting and enabling Docker"
+systemctl start docker
+systemctl enable docker
+
+echo "Creating database container "
+
+docker build -t alfred .
+
+docker run -d -p 3307:3307 --restart always --name ALFRED -v /var/lib/mysql:/var/lib/mysql alfred
+
+result=$( docker ps -a |grep documentsrv_test  )
+
+if [[ -n "$result" ]]; then
+  echo "removing container"
+  docker rm -f ALFRED
+  sleep 3
+
+  docker run -d -p 3307:3307 --restart always --name ALFRED -v /var/lib/mysql:/var/lib/mysql alfred
+else
+  echo "No such container"
+fi
+
+docker exec -it ALFRED mysql -u root -e "$(cat ./db.sql)"
+
+
+
+	
     * Start a container using the image you've craeted above with your Dockerfile - this container must run as follows:
         * It must be named ALFRED.
         * It must mount /var/lib/mysql to the host operating system to /var/lib/mysql.
         * It must mount /BATCAVE to the host operating system.
     * Checks whether a container exists called ALFRED and if it does, removes an recreates it.
     * Create a schema in the database called "wayneindustries" with one table in it called "fox" with columns "ID" and "Name".
     * Insert an entry with ID "50" and Name "BATMOBILE".
+	
+	CREATE DATABASE wayneindustries;
+
+	USE wayneindustries;
+
+	CREATE TABLE fox (ID INT NOT NULL, Name VARCHAR(40));
+
+	INSERT INTO fox (ID, Name) VALUES ('50', 'BATMOBILE')
+	
+	
 * Create an encrypted file called "secret" in the root of this repository that contains the root password of the database (the password must be "thisisadatabasepassword123456789!").
 * Change your Bash script to start the conainer using the root password from the "secret" file.
 * Commit and push your changes.
 
 # General
 * How would you ensure any change made to this Dockerfile is source controlled, approved, tested and deployed. Explain which tools you will use as if this was going into a production environment.
-* Commit and push your changes.
\ No newline at end of file
+* Commit and push your changes.
+
```

Comment:
I would use CI/CD tools jenkins pipelines, git and gitOps (Argo CD) for version control